### PR TITLE
 Restructuring docker files to build images per version

### PR DESCRIPTION
**Pull request by johnugeorge**
Dockerfiles are copied to each version so that each version is built independently.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/katib/527)

### CODE REVIEW

**Comment by johnugeorge** [comment]
Some changes are not added to avoid merge conflicts because of on going PR
1. UI folder changes are not added  #486 
2. envelopenet folder changes are not added.  #492 

@gaocegege @andreyvelich

**Comment by gaocegege** [comment]
Gotcha, thanks. I will do it for envelopenet

File: cmd/katib-controller/v1alpha2/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+ADD . /go/src/github.com/kubeflow/katib
+
+WORKDIR /go/src/github.com/kubeflow/katib/cmd/katib-controller
+# Build
+RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -a -o katib-controller.v1alpha2  ./v1alpha2
+# Copy the controller-manager into a thin image
+FROM alpine:3.7
+WORKDIR /app
+COPY --from=build-env /go/src/github.com/kubeflow/katib/cmd/katib-controller/katib-controller.v1alpha2 .
+ENTRYPOINT ["./katib-controller.v1alpha2"]
```
Comment (hougangliu) [inline cmd/katib-controller/v1alpha2/Dockerfile]: maybe s/katib-controller.v1alpha2/katib-controller? since the released images will have "v1alpha2"-like

Comment (johnugeorge) [reply cmd/katib-controller/v1alpha2/Dockerfile]: done

File: cmd/manager/v1alpha2/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+ADD . /go/src/github.com/kubeflow/katib
+WORKDIR /go/src/github.com/kubeflow/katib/cmd/manager
+RUN go build -o katib-manager.v1alpha2 ./v1alpha2
+
+FROM alpine:3.7
+WORKDIR /app
+RUN GRPC_HEALTH_PROBE_VERSION=v0.2.0 && \
+    wget -qO/bin/grpc_health_probe https://github.com/grpc-ecosystem/grpc-health-probe/releases/download/${GRPC_HEALTH_PROBE_VERSION}/grpc_health_probe-linux-amd64 && \
+    chmod +x /bin/grpc_health_probe
+COPY --from=build-env /go/src/github.com/kubeflow/katib/cmd/manager/katib-manager /app/
```
Comment (richardsliu) [inline cmd/manager/v1alpha2/Dockerfile]: This should be katib-manager.v1alpha2.

File: cmd/katib-controller/v1alpha2/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+# Copy in the go src
+ADD . /go/src/github.com/kubeflow/katib
+
+WORKDIR /go/src/github.com/kubeflow/katib/cmd/katib-controller
+# Build
+RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -a -o katib-controller.v1alpha2  ./v1alpha2
+# Copy the controller-manager into a thin image
+FROM alpine:3.7
+WORKDIR /app
+COPY --from=build-env /go/src/github.com/kubeflow/katib/cmd/katib-controller/katib-controller .
```
Comment (richardsliu) [inline cmd/katib-controller/v1alpha2/Dockerfile]: This should be katib-controller.v1alpha2. Or we can remove the ".v1alpha2" from line 9 above.

File: cmd/manager-rest/v1alpha2/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,10 @@
+FROM golang:alpine AS build-env
+# The GOPATH in the image is /go.
+ADD . /go/src/github.com/kubeflow/katib
+WORKDIR /go/src/github.com/kubeflow/katib/cmd/manager-rest
+RUN go build -o katib-manager-rest.v1alpha2 ./v1alpha2
+
+FROM alpine:3.7
+WORKDIR /app
+COPY --from=build-env /go/src/github.com/kubeflow/katib/cmd/manager-rest/katib-manager-rest /app/
```
Comment (richardsliu) [inline cmd/manager-rest/v1alpha2/Dockerfile]: katib-manager-rest.v1alpha2

**Comment by andreyvelich** [comment]
@johnugeorge Should I make the same with Dockerfiles for the UI?

**Comment by richardsliu** [comment]
@andreyvelich I think we should follow the same conventions for all components. That way we can easily find all v1alpha2 related images.

**Comment by andreyvelich** [comment]
@richardsliu Alright, will make changes in the UI PR.

**Comment by gaocegege** [comment]
Please rebase to remove dep-ensure step in CI

**Comment by richardsliu** [comment]
Also need to update the python script paths in `test/scripts/v1alpha2/python-tests.sh`.

**Pull request merged**